lectures/arellano.md

```diff
@@ -365,55 +366,43 @@ The output process is discretized using a [quadrature method due to Tauchen](htt
 
-We define a class that will store parameters, grids and transition
+We define a namedtuple to store parameters, grids and transition
 probabilities.
 
 ```{code-cell} ipython3
-:hide-output: false
+Arellano_Economy = namedtuple('Arellano_Economy', ('β', 'γ', 'r', 'ρ', 'η', 'θ', \
+                                                  'B_size', 'y_size', \
+                                                  'P', 'B_grid', 'y_grid', 'def_y'))
+```
 
-class Arellano_Economy:
-    " Stores data and creates primitives for the Arellano economy. "
-
-    def __init__(self,
-            B_grid_size=251,   # Grid size for bonds
-            B_grid_min=-0.45,   # Smallest B value
-            B_grid_max=0.45,    # Largest B value
-            y_grid_size=51,     # Grid size for income
-            β=0.953,            # Time discount parameter
-            γ=2.0,              # Utility parameter
-            r=0.017,            # Lending rate
-            ρ=0.945,            # Persistence in the income process
-            η=0.025,            # Standard deviation of the income process
-            θ=0.282,            # Prob of re-entering financial markets
-            def_y_param=0.969): # Parameter governing income in default
-
-        # Save parameters
-        self.β, self.γ, self.r, = β, γ, r
-        self.ρ, self.η, self.θ = ρ, η, θ
-
-        # Set up grids
-        self.y_grid_size = y_grid_size
-        self.B_grid_size = B_grid_size
-        B_grid = jnp.linspace(B_grid_min, B_grid_max, B_grid_size)
-        mc = qe.markov.tauchen(y_grid_size, ρ, η)
-        y_grid, P = jnp.exp(mc.state_values), mc.P
-
-        # Put grids on the device
-        self.B_grid = jax.device_put(B_grid)
-        self.y_grid = jax.device_put(y_grid)
-        self.P = jax.device_put(P)
-
-        # Output received while in default, with same shape as y_grid
-        self.def_y = jnp.minimum(def_y_param * jnp.mean(self.y_grid), self.y_grid)
-
-    def params(self):
-        return self.β, self.γ, self.r, self.ρ, self.η, self.θ
-
-    def sizes(self):
-        return self.B_grid_size, self.y_grid_size
-
-    def arrays(self):
-        return self.P, self.B_grid, self.y_grid, self.def_y
+```{code-cell} ipython3
+def create_arellano(B_size=251,   # Grid size for bonds
+                    B_min=-0.45,   # Smallest B value
+                    B_max=0.45,    # Largest B value
+                    y_size=51,     # Grid size for income
+                    β=0.953,            # Time discount parameter
+                    γ=2.0,              # Utility parameter
+                    r=0.017,            # Lending rate
+                    ρ=0.945,            # Persistence in the income process
+                    η=0.025,            # Standard deviation of the income process
+                    θ=0.282,            # Prob of re-entering financial markets
+                    def_y_param=0.969): # Parameter governing income in default
```

Comment:
Move the comments on parameters to where we define the `namedtuple`.

Reply:
Thanks @HumphreyYang , I will add the comments to both places, given the comments from another PR.